### PR TITLE
[PFX-602] - Typescript support

### PR DESCRIPTION
**Pull request by mmason2-godaddy**
## Summary
- Fix TypeScript support for Gasket apps
- Adjust types in generated files
- Tune the generated tsconfig
- TypeScript apps are now `type:module`
- Use `.js` extensions in TypeScript files - strange I know but it works
- Add [doc](https://github.com/godaddy/gasket/tree/PFX-602/packages/gasket-plugin-typescript#typescript--esm) for the `.js` extensions
- Drop `nodemon` and `ts-node` in favor of `tsx` - [npm tsx](https://www.npmjs.com/package/tsx)

## Local Testing
- Pull branch
- `cd packages/create-gasket-app` -> `npm i -g .`
- Create api preset app and select TS - need to use `preset-path` and set the TS plugin dependency to the local version
- Need to install the changes from the branch in the newly created app
  - `npm i <local/@gasket/core>` - Fixes a type issue for the gasket config in `makeGasket`
  -  `npm i <local/@gasket/plugin-express>` - Add additional types for the `./routes/index.js` file
- `npm run build` & `npm run start` - app should work

## Changelog
- Fix TypeScript support for Gasket apps

## Test Plan
Update tests and types.

### CODE REVIEW

File: packages/gasket-plugin-typescript/generator/tsconfig.json [hunk 1]
```diff
@@ -1,28 +1,31 @@
 {
+  "$schema": "https://json.schemastore.org/tsconfig",
+  "extends": [
+    "@tsconfig/node20/tsconfig.json"
```
Comment (agerard-godaddy) [inline packages/gasket-plugin-typescript/generator/tsconfig.json:4]: Seems like you're overriding most of what's being extended from this config. Either don't or avoid requiring this extra config package.

File: packages/gasket-plugin-typescript/lib/index.js [hunk 2]
```diff
@@ -8,15 +8,15 @@ module.exports = {
       const { pkg, files } = context;
 
       pkg.add('devDependencies', {
-        'nodemon': devDependencies.nodemon,
-        'ts-node': devDependencies['ts-node'],
+        '@tsconfig/node20': devDependencies['@tsconfig/node20'],
```
Comment (agerard-godaddy) [inline packages/gasket-plugin-typescript/lib/index.js:11]: Do not need this now:
```suggestion
```

File: packages/gasket-plugin-typescript/lib/index.js [hunk 3]
```diff
@@ -8,15 +8,14 @@ module.exports = {
...
-        'typescript': devDependencies.typescript
+        tsx: devDependencies.tsx,
+        typescript: devDependencies.typescript
       });
 
       pkg.add('scripts', {
         build: 'tsc',
         start: 'node dist/server.js',
-        local: 'nodemon server.ts'
+        local: 'tsx watch server.ts'
```
Comment (agerard-godaddy) [inline packages/gasket-plugin-typescript/lib/index.js:18]: Let's default to the `local` Gasket env here (test will need updated too)
```suggestion
        local: 'GASKET_ENV=local tsx watch server.ts'
```

**Review by agerard-godaddy — APPROVED** [review]
small tweak, then 👍

**Review by jpina1-godaddy — APPROVED** [review]

**Pull request merged**